mirte_telemetrix/scripts/mappings/pcb.py

```diff
@@ -86,7 +59,70 @@ def get_max_pwm_value():
     return board_mapping.get_max_pwm_value()
 
 
+def generate_motor_mapping(pin_a, pin_b):
+    # pin a has preference for pwm
+    # This will make sure, dp, pp and dd is always possible when using a connector without knowing what type of control
```

Comment:
Not really sure that problem you are solving here. Could you elaborate a bit more in the comments.

Reply:
The idea: with a connector name, we want to give all the info for the multiple types of motor controllers. One controller uses d1 and p1, other uses p1, p2 and the third (not supported by connectors) is d1, d2 and p1. 

This function generates { p1:A, p2:B, d1:B }, where A&B are the io pins. This makes it possible to be used with either of the two connector-supported motor (sw) drivers without knowing which driver.

Reply:
I've added some more comments